### PR TITLE
A basic implementation of Jackson reader that takes 1 file with multiple records to compliment the existing JacksonRecordReader

### PR DESCRIPTION
**Pull request by woolfel**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

Please review
https://github.com/deeplearning4j/deeplearning4j/blob/master/CONTRIBUTING.md before opening a pull request.

### CODE REVIEW

**Comment by AlexDBlack** [comment]
This looks great at first glance - is it ready for a review (and possible merge), or are more commits expected?

**Comment by woolfel** [comment]
I'm done with the javadocs for now. I will think about adding more unit tests tonight.

**Comment by woolfel** [comment]
I couldn't think of any other unit test I wanted to add, so I think it's good for now.

**Comment by woolfel** [comment]
I can take a look later this week. Feeling a little under the weather at the moment, since my son decided to give me his cold.

File: README.md [hunk 1]
```diff
@@ -110,3 +110,11 @@ mapping:
    [using the INtellij plugin](https://github.com/HPI-Information-Systems/Metanome/wiki/Installing-the-google-styleguide-settings-in-intellij-and-eclipse).
 
 6. Send a pull request, and bug us on Gitter until it gets merged and published.
+
+## Eclipse Setup
+
+1. Downloading the latest jar from https://projectlombok.org/download
+2. Double click the jar to install the plugin for Eclipse
```
Comment (huitseeker) [inline README.md]: Or run `java -jar /path/to/lombok.jar` ...

File: datavec-api/src/main/java/org/datavec/api/records/reader/impl/jackson/JacksonLineRecordReader.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ * @author peter
+ *
+ */
+public class JacksonLineRecordReader extends LineRecordReader {
+
+    private static final TypeReference<Map<String, Object>> typeRef = new TypeReference<Map<String, Object>>() {};
+    private FieldSelection selection;
+    private ObjectMapper mapper;
+
+	public JacksonLineRecordReader(FieldSelection selection, ObjectMapper mapper) {
```
Comment (huitseeker) [inline datavec-api/src/main/java/org/datavec/api/records/reader/impl/jackson/JacksonLineRecordReader.java]: Annotations ? https://projectlombok.org/features/constructor

File: datavec-api/src/main/java/org/datavec/api/records/reader/impl/jackson/JacksonLineRecordReader.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                            value = (String) o;
+                        } else if (o instanceof Number) {
+                            value = o.toString();
+                        } else {
+                            throw new IllegalStateException(
+                                            "Expected to find String on path " + Arrays.toString(currPath) + ", found "
+                                                            + o.getClass() + " with value " + o);
+                        }
+                    } else {
+                        //Expect to get a map...
```
Comment (huitseeker) [inline datavec-api/src/main/java/org/datavec/api/records/reader/impl/jackson/JacksonLineRecordReader.java]: This one is surprising. In which case would we expect this? Does that folow the Hadoop convention?

Comment (woolfel) [reply datavec-api/src/main/java/org/datavec/api/records/reader/impl/jackson/JacksonLineRecordReader.java]: Good question, I copied that from JacksonRecordReader lines 222-226. I don't need that for my specific use case, but I kept it the same as JacksonRecordReader to be safe. Feel free to change that if you think it doesn't apply to the line reader.

**Comment by woolfel** [comment]
I added JacksonLineRecordReader to TestSerialization

**Comment by AlexDBlack** [comment]
@woolfel I'm not seeing that TestSerialization test change here - did it get pushed up correctly?
https://github.com/deeplearning4j/DataVec/pull/467/files

**Comment by woolfel** [comment]
oops, I forgot to make a new pull request. just did it. thanks for catching that.

**Pull request merged**